pkg/operation/botanist/component/resourcemanager/resource_manager.go

```diff
@@ -379,9 +379,22 @@ func (r *resourceManager) ensureDeployment(ctx context.Context) error {
 
 		deployment.Spec.Template = corev1.PodTemplateSpec{
 			ObjectMeta: metav1.ObjectMeta{
-				Labels: r.getDeploymentTemplateLabels(),
+				Labels: utils.MergeStringMaps(r.getDeploymentTemplateLabels(), r.getNetworkPolicyLabels()),
 			},
 			Spec: corev1.PodSpec{
+				Affinity: &corev1.Affinity{
+					PodAntiAffinity: &corev1.PodAntiAffinity{
+						PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
+							{
+								Weight: 100,
```

Comment:
Learning question:
We are selecting the maximum possible weight regarding the scheduling decision for the podAntiAffinity.
That means it is relatively important in regard to other weights, for example if we added additional [AntiAffinity rules or in relation to resource requests](https://github.com/kubernetes/kubernetes/blob/36e69a33030c16e19b5d7edba9f755cae917d09d/pkg/apis/core/types.go#L2542-L2544).
While we chose `100` here we chose `1` for the [kube-apiserver](https://github.com/gardener/gardener/blob/97e2e316891bfad3c7e381d7fea7d068f7453aa9/pkg/operation/botanist/component/kubeapiserver/deployment.go#L166) 
Why do we select different weights in those two components ?
Thanks!

Reply:
Good question. Honestly, I don't think there are dedicated reasons for these values. In my case, I just applied what we already do for GSAC (https://github.com/gardener/gardener/blob/29ffac5ed724458803b7748800e9de6f07c84f0a/pkg/operation/botanist/component/seedadmissioncontroller/seedadmissioncontroller.go#L211-L223).

I don't know why we use `1` for KAPI, I would guess that it was introduced based on the example provided in https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity (here, `1` is used as well).